modules/representation_management/app/models/representation_management/next_steps_email_data.rb

```diff
@@ -20,13 +19,84 @@ class NextStepsEmailData
     validates :first_name, presence: true
     validates :form_name, presence: true
     validates :form_number, presence: true
-    validates :representative_type, presence: true
-    validates :representative_type, inclusion: { in: AccreditedIndividual.individual_types.keys }
-    validates :representative_name, presence: true
-    validates :representative_address, presence: true
+    validates :entity_type, presence: true
+    validates :entity_id, presence: true
+    validate :entity_exists?
 
-    def representative_type_humanized
-      @representative_type_humanized ||= representative_type.humanize.titleize
+    def entity
+      @entity ||= find_entity
+    end
+
+    def entity_display_type
+      display_type =
+        if entity.is_a?(Veteran::Service::Representative)
+          veteran_service_representative_type
+        elsif entity.is_a?(AccreditedIndividual)
+          entity.individual_type
+        else
+          'Veteran Service Organization'
+        end
+      display_type.humanize.titleize
+    end
+
+    def entity_name
+      if entity_type == 'individual'
+        entity.full_name.strip
+      elsif entity_type == 'organization'
+        entity.name.strip
+      end
+    end
+
+    def entity_address
+      <<~ADDRESS.squish
+        #{entity.address_line1}
```

Comment:
Will null values error out? What if entity is null?

Reply:
Null values should result in an empty string.  The entity shouldn't be null by the time this is called, we're validating that `entity_exists?` before we call this method.

Reply:
@RachalCassity do you want me to add null checks for the values that may not exist, something like:
```ruby
<<~ADDRESS
  #{entity.address_line1}
  #{entity.address_line2 if entity.address_line2}
  #{entity.address_line3 if entity.address_line3}
  #{entity.city}, #{entity.state_code} #{entity.zip_code}
  #{entity.country_code_iso3}
ADDRESS
```
?

Reply:
And would adding those checks require an additional approval from one of my team members?

Reply:
> The entity shouldn't be null by the time this is called, we're validating that `entity_exists?` before we call this method.

Sounds good to me 👍